src/lando/api/views.py

```diff
@@ -63,6 +63,31 @@ def _wrapper(self, request, *args, **kwargs):  # noqa: ANN001
     return _wrapper
 
 
+def generate_warnings_and_blockers(
+    target_repo: Repo, pull_request: PullRequest, request: HttpRequest
+) -> dict[str, list[str]]:
+    """Run checks on a pull request and return blockers and warnings."""
+    patch_helper = PullRequestPatchHelper(pull_request)
+    author_email = pull_request.author[1]
+    landing_checks = LandingChecks(author_email)
+    checks = [
+        chk.name()
+        for chk in ALL_CHECKS
+        # This is checking for secure revisions in BMO. We skip this for now.
+        if chk.name() != BugReferencesCheck.name()
+    ]
+    blockers = landing_checks.run(
+        checks,
+        [patch_helper],
+    )
```

Comment:
nit: Ugh. I think I completely forgot to honour `Repo.hooks` in my original code.

Do you think we could/should fix this now?

```suggestion
    blockers = landing_checks.run(
        target_repo.hooks,
        [patch_helper],
    )
```